File: identify_ShotID.m
function identify_ShotID(ti, Data, ColumnName)
global SA

col_si = strcmp('ShotID', ColumnName);
col_mi = strcmp('Mirrored', ColumnName);

ShotID = cell2mat(Data(ti, col_si));
mirrored = cell2mat(Data(ti, col_mi));

[~, si, ~] = intersect(cell2mat(SA.Table.ShotID)+cell2mat(SA.Table.Mirrored)/10, ShotID+mirrored/10);

SA.Current_ti = ti;
SA.Current_ShotID = SA.Table.ShotID(si);
SA.Current_si = si;

File: identify_ShotId.py
import numpy as np

def identify_shot_id(ti, data, column_name, SA):
    col_si = column_name.index('ShotID')
    col_mi = column_name.index('Mirrored')

    shot_id = data[ti][col_si]
    mirrored = data[ti][col_mi]

    combined = np.array(SA['Table']['ShotID']) + np.array(SA['Table']['Mirrored']) / 10
    target = shot_id + mirrored / 10

    si = np.where(combined == target)[0][0]

    SA['Current_ti'] = ti
    SA['Current_ShotID'] = SA['Table']['ShotID'][si]
    SA['Current_si'] = si